#!/usr/bin/env python

PACKAGE = "pwm_servo"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add( "trim1", double_t, 0, "Trim of servo1 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim2", double_t, 0, "Trim of servo2 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim3", double_t, 0, "Trim of servo3 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim4", double_t, 0, "Trim of servo4 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim5", double_t, 0, "Trim of servo5 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim6", double_t, 0, "Trim of servo6 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim7", double_t, 0, "Trim of servo7 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim8", double_t, 0, "Trim of servo8 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim9", double_t, 0, "Trim of servo9 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim10", double_t, 0, "Trim of servo10 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim11", double_t, 0, "Trim of servo11 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim12", double_t, 0, "Trim of servo12 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim13", double_t, 0, "Trim of servo13 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim14", double_t, 0, "Trim of servo14 (degree)", 0.0, -45.0, 45.0 )
gen.add( "trim15", double_t, 0, "Trim of servo15 (degree)", 0.0, -45.0, 45.0 )

gen.add( "limitMax1", double_t, 0, "Max limit of servo1 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin1", double_t, 0, "Min limit of servo1 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax2", double_t, 0, "Max limit of servo2 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin2", double_t, 0, "Min limit of servo2 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax3", double_t, 0, "Max limit of servo3 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin3", double_t, 0, "Min limit of servo3 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax4", double_t, 0, "Max limit of servo4 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin4", double_t, 0, "Min limit of servo4 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax5", double_t, 0, "Max limit of servo5 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin5", double_t, 0, "Min limit of servo5 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax6", double_t, 0, "Max limit of servo6 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin6", double_t, 0, "Min limit of servo6 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax7", double_t, 0, "Max limit of servo7 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin7", double_t, 0, "Min limit of servo7 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax8", double_t, 0, "Max limit of servo8 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin8", double_t, 0, "Min limit of servo8 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax9", double_t, 0, "Max limit of servo9 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin9", double_t, 0, "Min limit of servo9 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax10", double_t, 0, "Max limit of servo10 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin10", double_t, 0, "Min limit of servo10 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax11", double_t, 0, "Max limit of servo11 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin11", double_t, 0, "Min limit of servo11 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax12", double_t, 0, "Max limit of servo12 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin12", double_t, 0, "Min limit of servo12 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax13", double_t, 0, "Max limit of servo13 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin13", double_t, 0, "Min limit of servo13 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax14", double_t, 0, "Max limit of servo14 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin14", double_t, 0, "Min limit of servo14 (degree)", 0.0, 0.0, 180.0 )
gen.add( "limitMax15", double_t, 0, "Max limit of servo15 (degree)", 180.0, 0.0, 180.0 )
gen.add( "limitMin15", double_t, 0, "Min limit of servo15 (degree)", 0.0, 0.0, 180.0 )

exit( gen.generate(PACKAGE, "pwm_servo", "servo_param") )
